### PR TITLE
Get rid of `recommonmark` dependency 

### PR DESCRIPTION
**Pull request by achiefa**
This PR addresses #1978.

### CODE REVIEW

**Comment by RoyStegeman** [comment]
Thanks a lot for starting this! Don't forget to also delete the markdown file you're replacing.

**Comment by achiefa** [comment]
> Don't forget to also delete the markdown file you're replacing.

Yes, I was't sure whether I should've deleted them or not. I was waiting for your feedback, many thanks!

**Comment by achiefa** [comment]
I'm currently working on this PR. I just want to mention that nested inline markups are currently not possible in reStructuredText, as explained [here](https://docutils.sourceforge.io/FAQ.html#is-nested-inline-markup-possible). There are some workarounds, but they are really ugly and are not worth the effort. Also, the documentation discourages to adopt them. This just to say that, in covering *md* to *rst*, I'm not using nested inline markups when dealing with cross-references.

**Comment by RoyStegeman** [comment]
thanks, it's a bit ugly anyway so that's fine.

**Comment by achiefa** [comment]
Ok, I think I'm done. Please, @RoyStegeman and @scarlehoff could you please check that I didn't forget anything?

**Comment by RoyStegeman** [comment]
Thanks! I'll have a look next week.

**Comment by RoyStegeman** [comment]
Could you also remove it from the dependencies?

**Comment by achiefa** [comment]
I only see dependencies on `recommonmark`  in `pyproject.toml` and in `meta.yaml` in the conda-recipe. Should I just remove  any `recommonmark` key from these files?

**Comment by RoyStegeman** [comment]
Yes exactly, I suspect you will also have to remove the recommonmark settings from some config file if you haven't already.

File: doc/sphinx/source/conf.py
Comment (achiefa) [inline doc/sphinx/source/conf.py:1]: Should I also remove the `setup` function at the very end of this?

Comment (RoyStegeman) [reply doc/sphinx/source/conf.py:1]: I think you can, at least the recommmonmark parts but I think the entire function. If you delete it and can still build the docs than its fine

**Comment by achiefa** [comment]
Ok, the doc compiles even after that. Here you can find the log of the build (both stdout and stderr).
[build.log](https://github.com/user-attachments/files/17365054/build.log)

**Review by RoyStegeman — APPROVED** [review]
Thanks, looks good! 

At some point we should also take care of the (important) warning such as non-existent labels https://github.com/NNPDF/nnpdf/issues/1360

**Pull request merged**